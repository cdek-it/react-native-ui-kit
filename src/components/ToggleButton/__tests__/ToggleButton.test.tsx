import { IconArrowDownRight } from '@tabler/icons-react-native'
import { render, userEvent } from '@testing-library/react-native'

import {
  ToggleButton,
  type ToggleButtonProps,
  ToggleButtonTestId,
} from '../ToggleButton'

describe('ToggleButton', () => {
  const defaultProps: ToggleButtonProps = {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    onPress: () => {},
  }

  describe('snapshots', () => {
    const snapshotCases: Array<[string, Partial<ToggleButtonProps>]> = [
      ['default', {}],
      [
        'checked = true, disabled = false, iconOnly = false, size = xlarge, with style, with Icon',
        {
          checked: true,
          disabled: false,
          iconOnly: false,
          size: 'xlarge',
          style: { margin: 10 },
          Icon: IconArrowDownRight,
        },
      ],
      [
        'checked = false, disabled = true, iconOnly = false, size = large, with Icon, with label',
        {
          checked: false,
          disabled: true,
          iconOnly: false,
          size: 'large',
          Icon: IconArrowDownRight,
          label: 'ButtonToggle',
        },
      ],
      [
        'checked = false, disabled = false, iconOnly = true, size = base, with Icon, with label',
        {
          checked: false,
          disabled: false,
          iconOnly: true,
          size: 'base',
          Icon: IconArrowDownRight,
          label: 'ButtonToggle',
        },
      ],
      [
        'checked = false, disabled = false, iconOnly = false, size = small, with Icon, with label, iconPos = left',
        {
          checked: false,
          disabled: false,
          iconOnly: false,
          size: 'small',
          Icon: IconArrowDownRight,
          label: 'ButtonToggle',
          iconPos: 'left',
        },
      ],
      [
        'checked = false, disabled = false, iconOnly = false, size = small, with Icon, with label, iconPos = right',
        {
          checked: false,
          disabled: false,
          iconOnly: false,
          size: 'small',
          Icon: IconArrowDownRight,
          label: 'ButtonToggle',
          iconPos: 'right',
        },
      ],
    ]

    beforeAll(() => {
      jest.mock(
        'react-native/Libraries/Components/Pressable/Pressable',
        () => ({ default: 'Pressable' })
      )
    })

    afterAll(() => {
      jest.unmock('react-native/Libraries/Components/Pressable/Pressable')
    })

    test.each(snapshotCases)('%s', (_, props) => {
      const { toJSON } = render(<ToggleButton {...defaultProps} {...props} />)

      expect(toJSON()).toMatchSnapshot()
    })
  })

  test('should handle press', async () => {
    const mockedOnPress = jest.fn()
    const { queryByTestId } = render(<ToggleButton onPress={mockedOnPress} />)
    const pressable = queryByTestId(ToggleButtonTestId.root)
    const user = userEvent.setup()

    await user.press(pressable)

    expect(mockedOnPress).toHaveBeenCalled()
  })

  test('should NOT handle press', async () => {
    const mockedOnPress = jest.fn()
    const { queryByTestId } = render(
      <ToggleButton disabled onPress={mockedOnPress} />
    )
    const pressable = queryByTestId(ToggleButtonTestId.root)
    const user = userEvent.setup()

    await user.press(pressable)

    expect(mockedOnPress).not.toHaveBeenCalled()
  })
})
