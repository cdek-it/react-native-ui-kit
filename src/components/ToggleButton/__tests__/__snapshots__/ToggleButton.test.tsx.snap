// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton snapshots checked = false, disabled = false, iconOnly = false, size = small, with Icon, with label, iconPos = left 1`] = `
<Pressable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      undefined,
      [
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        false,
        false,
        false,
      ],
    ]
  }
  testID="ToggleButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 100,
          "flexDirection": "row",
          "gap": 10.5,
          "paddingHorizontal": 21,
          "paddingVertical": 0,
        },
        {
          "gap": 7,
          "minHeight": 28,
          "paddingHorizontal": 10.5,
        },
        false,
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
          false,
          false,
          false,
          false,
        ],
      ]
    }
    testID="ToggleButton.container"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={14}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={14}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            false,
          ],
          {
            "flex": 0,
            "height": 14,
            "width": 14,
          },
        ]
      }
      testID="ToggleButton.icon"
      vbHeight={24}
      vbWidth={24}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        color="rgba(0, 0, 0, 0.6000)"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M7 7l10 10"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M17 8l0 9l-9 0"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        [
          {
            "flexShrink": 1,
          },
          [
            {
              "fontWeight": "700",
              "includeFontPadding": false,
              "verticalAlign": "middle",
            },
            {
              "fontSize": 12.25,
            },
          ],
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            false,
          ],
        ]
      }
      testID="ToggleButton.text"
    >
      ButtonToggle
    </Text>
  </View>
</Pressable>
`;

exports[`ToggleButton snapshots checked = false, disabled = false, iconOnly = false, size = small, with Icon, with label, iconPos = right 1`] = `
<Pressable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      undefined,
      [
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        false,
        false,
        false,
      ],
    ]
  }
  testID="ToggleButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 100,
          "flexDirection": "row",
          "gap": 10.5,
          "paddingHorizontal": 21,
          "paddingVertical": 0,
        },
        {
          "gap": 7,
          "minHeight": 28,
          "paddingHorizontal": 10.5,
        },
        false,
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
          false,
          false,
          false,
          false,
        ],
      ]
    }
    testID="ToggleButton.container"
  >
    <Text
      style={
        [
          {
            "flexShrink": 1,
          },
          [
            {
              "fontWeight": "700",
              "includeFontPadding": false,
              "verticalAlign": "middle",
            },
            {
              "fontSize": 12.25,
            },
          ],
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            false,
          ],
        ]
      }
      testID="ToggleButton.text"
    >
      ButtonToggle
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={14}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={14}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            false,
          ],
          {
            "flex": 0,
            "height": 14,
            "width": 14,
          },
        ]
      }
      testID="ToggleButton.icon"
      vbHeight={24}
      vbWidth={24}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        color="rgba(0, 0, 0, 0.6000)"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M7 7l10 10"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M17 8l0 9l-9 0"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</Pressable>
`;

exports[`ToggleButton snapshots checked = false, disabled = false, iconOnly = true, size = base, with Icon, with label 1`] = `
<Pressable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      undefined,
      [
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        false,
        false,
        false,
      ],
    ]
  }
  testID="ToggleButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 100,
          "flexDirection": "row",
          "gap": 10.5,
          "paddingHorizontal": 21,
          "paddingVertical": 0,
        },
        {
          "gap": 7,
          "minHeight": 35,
          "paddingHorizontal": 14,
        },
        {
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "paddingVertical": 0,
        },
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
          false,
          false,
          false,
          false,
        ],
      ]
    }
    testID="ToggleButton.container"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={17.5}
      bbWidth={17.5}
      fill="none"
      focusable={false}
      height={17.5}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            false,
          ],
          {
            "flex": 0,
            "height": 17,
            "width": 17,
          },
        ]
      }
      testID="ToggleButton.icon"
      vbHeight={24}
      vbWidth={24}
      width={17.5}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        color="rgba(0, 0, 0, 0.6000)"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M7 7l10 10"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M17 8l0 9l-9 0"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</Pressable>
`;

exports[`ToggleButton snapshots checked = false, disabled = true, iconOnly = false, size = large, with Icon, with label 1`] = `
<Pressable
  disabled={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      undefined,
      [
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        false,
        false,
        {
          "borderColor": "#dbdada",
          "opacity": 0.6,
        },
      ],
    ]
  }
  testID="ToggleButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 100,
          "flexDirection": "row",
          "gap": 10.5,
          "paddingHorizontal": 21,
          "paddingVertical": 0,
        },
        {
          "minHeight": 49,
        },
        false,
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
          false,
          false,
          false,
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
        ],
      ]
    }
    testID="ToggleButton.container"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={21}
      bbWidth={21}
      fill="none"
      focusable={false}
      height={21}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
          ],
          {
            "flex": 0,
            "height": 21,
            "width": 21,
          },
        ]
      }
      testID="ToggleButton.icon"
      vbHeight={24}
      vbWidth={24}
      width={21}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        color="rgba(0, 0, 0, 0.6000)"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M7 7l10 10"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
        <RNSVGPath
          color="rgba(0, 0, 0, 0.6000)"
          d="M17 8l0 9l-9 0"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        [
          {
            "flexShrink": 1,
          },
          [
            {
              "fontWeight": "700",
              "includeFontPadding": false,
              "verticalAlign": "middle",
            },
            {
              "fontSize": 17.5,
            },
          ],
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            false,
            false,
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
          ],
        ]
      }
      testID="ToggleButton.text"
    >
      ButtonToggle
    </Text>
  </View>
</Pressable>
`;

exports[`ToggleButton snapshots checked = true, disabled = false, iconOnly = false, size = xlarge, with style, with Icon 1`] = `
<Pressable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      {
        "margin": 10,
      },
      [
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        false,
      ],
    ]
  }
  testID="ToggleButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 100,
          "flexDirection": "row",
          "gap": 10.5,
          "paddingHorizontal": 21,
          "paddingVertical": 0,
        },
        {
          "minHeight": 56,
        },
        {
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "paddingVertical": 0,
        },
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
          false,
          {
            "backgroundColor": "#188700",
          },
          false,
          false,
        ],
      ]
    }
    testID="ToggleButton.container"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={31.5}
      bbWidth={31.5}
      fill="none"
      focusable={false}
      height={31.5}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            {
              "color": "rgba(0, 0, 0, 0.6000)",
            },
            false,
            {
              "color": "#ffffff",
            },
            false,
            false,
          ],
          {
            "flex": 0,
            "height": 31,
            "width": 31,
          },
        ]
      }
      testID="ToggleButton.icon"
      vbHeight={24}
      vbWidth={24}
      width={31.5}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        color="#ffffff"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          color="#ffffff"
          d="M7 7l10 10"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
        <RNSVGPath
          color="#ffffff"
          d="M17 8l0 9l-9 0"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="ToggleButton.icon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</Pressable>
`;

exports[`ToggleButton snapshots default 1`] = `
<Pressable
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      undefined,
      [
        {
          "borderColor": "rgba(255, 255, 255, 0.0001)",
        },
        false,
        false,
        false,
        false,
      ],
    ]
  }
  testID="ToggleButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 100,
          "flexDirection": "row",
          "gap": 10.5,
          "paddingHorizontal": 21,
          "paddingVertical": 0,
        },
        {
          "gap": 7,
          "minHeight": 35,
          "paddingHorizontal": 14,
        },
        {
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "paddingVertical": 0,
        },
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.1000)",
          },
          false,
          false,
          false,
          false,
        ],
      ]
    }
    testID="ToggleButton.container"
  />
</Pressable>
`;
